databricks_cli/permissions/api.py

```diff
@@ -0,0 +1,343 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from enum import Enum
+
+from databricks_cli.sdk.permissions_service import PermissionsService
+from .exceptions import PermissionsError
+
+
+class PermissionTargets(Enum):
+    clusters = 'clusters'
+    cluster = clusters
+    directories = 'directories'
+    directory = directories
+    instance_pools = 'instance-pools'
+    instance_pool = instance_pools
+    jobs = 'jobs'
+    job = jobs
+    notebooks = 'notebooks'
+    notebook = notebooks
+    registered_models = 'registered-models'
+    registered_model = registered_models
+    model = registered_models
+    models = registered_models
+
+    @classmethod
+    def values(cls):
+        return [e.value for e in PermissionTargets]
+
+    @classmethod
+    def help_values(cls):
+        return ', '.join([e.value for e in PermissionTargets])
+
+    @classmethod
+    def get(cls, item):
+        if '-' in item:
+            item = item.replace('-', '_')
+        return PermissionTargets[item]
+
+
+class PermissionLevel(Enum):
+    no_permissions = 'NONE'
+    manage = 'CAN_MANAGE'
+    manage_staging_versions = 'CAN_MANAGE_STAGING_VERSIONS'
+    manage_production_versions = 'CAN_MANAGE_PRODUCTION_VERSIONS'
+    restart = 'CAN_RESTART'
+    attach = 'CAN_ATTACH_TO'
+    manage_run = 'CAN_MANAGE_RUN'
+    owner = 'IS_OWNER'
+    view = 'CAN_VIEW'
+    read = 'CAN_READ'
+    run = 'CAN_RUN'
+    edit = 'CAN_EDIT'
+    use = 'CAN_USE'
+
+    @classmethod
+    def names(cls):
+        return [e.name for e in PermissionLevel]
+
+    @classmethod
+    def values(cls):
+        return [e.value for e in PermissionLevel]
+
+    @classmethod
+    def help_values(cls):
+        return ', '.join([e.value for e in PermissionLevel])
+
+
+class BasicPermissions(object):
```

Comment:
it looks over-engineered. please simplify. 

technically, it all can be replaced by single simple lookup structure.